src/SolidStateDetector/DetectorGeometries.jl

```diff
@@ -407,15 +411,86 @@ function set_pointtypes_and_fixed_potentials!(pointtypes::Array{PointType, N}, p
         else
             contact.id == weighting_potential_contact_id ? 1 : 0
         end
-        contact_gridpoints = paint_object(contact, grid)
-        for gridpoint in contact_gridpoints
-            potential[ gridpoint... ] = pot
-            pointtypes[ gridpoint... ] = zero(PointType)
+        # contact_gridpoints = paint_object(contact, grid)
+        # for gridpoint in contact_gridpoints
+        #     potential[ gridpoint... ] = pot
+        #     pointtypes[ gridpoint... ] = zero(PointType)
+        # end
+        fs = ConstructiveSolidGeometry.surfaces(contact.geometry)
+        for face in fs
+            paint!(pointtypes, potential, face, contact.geometry, pot, grid)
         end
     end
     nothing
 end
 
+function paint!(pointtypes, potential, face::ConstructiveSolidGeometry.Polygon, geometry, pot_value, grid::CartesianGrid)
```

Comment:
This painting algorithm should go into a separate file

Reply:
I agree. As I was not sure what function I will need at the beginning I just added it there. At the end we can do some clean up.